File: nim/sum-of-multiples/sum_of_multiples.nim
from std/sequtils import deduplicate
from math import sum

proc sum*(limit: int, nums: seq[int]): int =
    var valid_multiples: seq[int]
    for number in nums:
        if number <= 0: continue
        var mult: int = number
        while mult < limit:
            valid_multiples.add(mult)
            mult += number
    sum(deduplicate(valid_multiples))

File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  def initialize(*values)
    @values = values.reject { |val| val.zero? }
  end

  def to(limit)
    multiples = []
    @values.each do |val|
      mult = val
      while mult < limit
        multiples << mult
        mult += val
      end
    end
    multiples.uniq.sum
  end
end